autogen/cluster.tf

```diff
@@ -219,14 +219,31 @@ resource "google_container_cluster" "primary" {
 /******************************************
   Create Container Cluster node pools
  *****************************************/
+resource "random_id" "name" {
+  # if any node_pool definition has a create_before_destroy key, then create random_id names
```

Comment:
Generating the random_id is a safe operation, so I think we could simply drop this `count` and simplify the code.

Reply:
Sounds good. I will update `count`. 

My original thinking was that if one were to see the `random_id.name` changing and there are no node pools with `create_before_destroy` defined, it would be confusing to the user of this module because they might think something is changing when in fact nothing is changing for them or their node pools.